internal/npm_install/generate_build_file.ts

```diff
@@ -571,52 +597,67 @@ export function getDirectDependencySet(pkgJsonPath: string): Set<string> {
 /**
  * Finds and returns an array of all packages under a given path.
  */
-function findPackages(p: string, dependencies: Set<string>) {
-  if (!isDirectory(p)) {
-    return [];
-  }
-
+async function findPackages(p: string, dependencies: Set<string>): Promise<Dep[]> {
   const pkgs: Dep[] = [];
+  const _findPackages = async (p: string) => {
```

Comment:
I *think* you can avoid the inner functions here and in `listFiles` basically doing the same as before your change, using `...` on the recursive call? Or was there another reason for the inner function?

```
  if (!await isDirectory(p)) {
    return [];
  }
  
  const pkgs: Dep[] = [];
  const listing = await fs.readdir(p);

  await Promise.all(listing.map(async f => {
    // filter out folders such as `.bin` which can create
    // issues on Windows since these are "hidden" by default
    if (f.startsWith('.')) return [];
    const pf = path.posix.join(p, f);
    
    if (await isDirectory(pf)) {
      if (f.startsWith('@')) {
        await findPackages(pf, dependencies);
      } else {
        pkgs.push(
          await parsePackage(pf, dependencies),
          ... await findPackages(path.posix.join(pf, 'node_modules'), dependencies)
        );
      }
    }
  }));
```

Reply:
I do this before I open this PR.
the problem is the `pkgs` need to be sort to ensure the order, So it just needs one `pkgs` instance and sort once through all inner recursive calls. 

I think it would be neater to move `pkgs.sort` into findPackages.

Reply:
Ah I see, that's why you moved it out originally? Sorry I missed that, I was just trying to reduce the diff since I thought it was an unnecessary change. Looks like I was wrong there?

Reply:
You are right, It is an unnecessary change, I just put the sort call into findPackages and listFiles in 12c687a, Sorry, I mixed the semi and refactor commit, It may be confused.

Reply:
@cuyl did you want to try dropping these inner functions and putting the `.sort()` outside?

I know you did similar before so sorry if I'm flip-flopping here, I don't think I realized what was going on at first...

Reply:
OK, I have put the sort outside and rename the functions to more reasonable names.